### PR TITLE
Decomposedfs allow arbitrary length spaceid

### PR DESCRIPTION
**Pull request by butonic**
The decomposedfs no longer panics when trying to look up spaces with a non uuid length id.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -81,6 +81,7 @@ require (
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
 	golang.org/x/sys v0.0.0-20220307203707-22a9840ba4d7
 	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211
+	golang.org/x/text v0.3.7
```
Comment (kobergj) [inline go.mod:84]: Is this change intended? Where is this new dependency coming from?

**Review by kobergj — CHANGES_REQUESTED** [review]

**Review by kobergj — APPROVED** [review]

File: pkg/storage/utils/decomposedfs/spaces.go [hunk 2]
```diff
@@ -215,20 +215,21 @@ func (fs *Decomposedfs) canCreateSpace(ctx context.Context, spaceID string) bool
...
-	// ....spaces4c510ada-c86b-4815-8820-42cdf82c3d51nodes4c510ada-c86b-4815-8820-42cdf82c3d51
-	if link[0:10] != "....spaces" || link[46:51] != "nodes" {
+	// ../../spaces/sp/ace-id/nodes/sh/or/tn/od/eid
+	// 0  1  2      3  4      5     6  7  8  9  10
+	parts := strings.Split(link, string(filepath.Separator))
+	if len(parts) != 11 || parts[0] != ".." || parts[1] != ".." || parts[2] != "spaces" || parts[5] != "nodes" {
 		return "", "", errtypes.InternalError("malformed link")
 	}
-	return link[10:46], link[51:], nil
+	return parts[3] + parts[4], parts[6] + parts[7] + parts[8] + parts[9] + parts[10], nil
```
Comment (C0rby) [inline pkg/storage/utils/decomposedfs/spaces.go:232]: I personally would do
```suggestion
	return strings.Join(parts[3:5], ""), strings.Join(parts[6:11], ""), nil
```

**Review by C0rby — APPROVED** [review]
Just a small nitpick.
